pkg/operator/controller/dns/controller.go

```diff
@@ -89,6 +93,15 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		log.V(2).Info("warning: dnsrecord is missing owner label", "dnsrecord", record, "ingresscontroller", ingressName)
 		return reconcile.Result{RequeueAfter: 1 * time.Minute}, nil
 	}
+
+	// Existing 4.1 records will have a zero TTL. Instead of making all the client implementations guard against
+	// zero TTLs, simply ignore the record until the TTL is updated by the ingresscontroller controller. Report
+	// this through events so we can detect problems with our migration.
+	if record.Spec.RecordTTL <= 0 {
+		r.recorder.Eventf(record, "Warning", "ZeroTTL", "Record is missing TTL and will be temporarily ignored; the TTL will be automatically updated and the record will be retried.")
+		return reconcile.Result{RequeueAfter: 15 * time.Second}, nil
```

Comment:
`RequeueAfter` should be unnecessary—dns_controller's watch on DNSRecord resources should trigger reconciliation when ingress_controller updates the TTL.